src/engraving/libmscore/articulation.h

```diff
@@ -122,6 +122,14 @@ class Articulation final : public EngravingItem
     };
     static AnchorGroup anchorGroup(SymId);
 
+    bool _isDouble = false;
+    bool _isTenuto = false;
+    bool _isStaccato = false;
+    bool _isAccent = false;
+    bool _isMarcato = false;
+    bool _isLuteFingering = false;
```

Comment:
To reduce memory usage, you _could_ store these using a single field of flags. But I'd first ask @vpereverzev if he thinks that's better.

Anyway, to do that, you would create an enum like this:
```c++
enum class ArticulationCategory: char { // char makes sure that this will use only 1 byte, 
                                        // which consists of 8 bits, 
                                        // which is enough to store these 6 flags
    None = 0x00
    Double = 0x01
    Tenuto = 0x02
    Staccato = 0x04
    Accent = 0x08
    Marcato = 0x10
    LuteFingering = 0x20
}
```
Then create a Flags from that enum: (you need to `#include "flags.h"` for this)
```c++
DECLARE_FLAGS(ArticulationCategories, ArticulationCategory)
DECLARE_OPERATORS_FOR_FLAGS(ArticulationCategories)
```

Then replace all these members with:
```suggestion
    ArticulationCategories m_categories = ArticulationCategory::None;
```

Then replace the `is...` methods with:
```c++
    bool isDouble() const { return m_categories & ArticulationCategory::Double; }
```
etc.

Then modify `computeCategories` like this:
```c++
m_categories.setFlag(ArticulationCategory::Double, 
                     _symId == SymId::articMarcatoStaccatoAbove || _symId == SymId::articMarcatoStaccatoBelow
                     || _symId == SymId::articTenutoStaccatoAbove || …)

Reply:
Thanks! I'll try!